diff --git a/include/libtpms/tpm_types.h b/include/libtpms/tpm_types.h
index 56c57822..0a69ebd8 100644
--- a/include/libtpms/tpm_types.h
+++ b/include/libtpms/tpm_types.h
@@ -52,8 +52,10 @@ extern "C" {
 
 /* 2.2.1 Basic data types rev 87 */
 #if !defined(TPM_HAVE_TPM2_DECLARATIONS)
+#ifndef BYTE
 typedef unsigned char  BYTE;            /* Basic byte used to transmit all character fields.  */
 #endif
+#endif
 typedef unsigned char  TPM_BOOL;        /* TRUE/FALSE field. TRUE = 0x01, FALSE = 0x00 Use TPM_BOOL
                                            because MS VC++ defines BOOL on Windows */
 
